### DISCUSSION TITLE
How to have only one retry topic for exponential retry when using RetryTopicConfiguration i.e nothing should be appended at the end like delay, index etc?

### DISCUSSION THREAD

**Discussion by hariraogotit** [spring-projects/spring-kafka#2639] [Q&A] [answered]
Team,

I have the following in MyRetryConfig.java that is annotated with @configuration

```
@Bean
 public RetryTopicConfiguration myRetryTopic(KafkaTemplate<String, byte[]> kafkaTemplate) {
   return RetryTopicConfigurationBuilder
           .newInstance()
           .exponentialBackoff(3000, 2, 100000)
           .maxAttempts(5)
           .retryTopicSuffix("-asl-nm.retry")
           .dltSuffix("-asl-nm.dlt")
           .create(kafkaTemplate);
 }
```

**Many retry topics** are created with index and delay appended. I went through the topic naming documentation - https://docs.spring.io/spring-kafka/reference/html/#topic-naming 

I see options to pick Index or Delay, https://docs.spring.io/spring-kafka/reference/html/#single-topic-maxinterval-delay, https://docs.spring.io/spring-kafka/reference/html/#custom-naming-strategies BUT I want to just have ONE RETRY TOPIC?

Eg my original topic is original-topic AND retry should only be original-topic-asl-nm.retry

Kindly let me  know how can I achieve this? Any leads will be helpful and much appreciated. Thank you.

**Comment by garyrussell** [comment, marked as answer]
You can't have only one topic for exponential retry; there is a recent change to use only one topic for retries once the max delay has been reached, but each individual retry before that one must be in their own topic; otherwise records ready to process will be blocked behind those that have longer delays.

**Reply by garyrussell** [reply]
See this answer for how to use a custom naming strategy: https://stackoverflow.com/questions/75723808/send-kafka-retryable-message-to-one-topic-via-spring-retry/75733513#75733513

**Reply by hariraogotit** [reply]
> You can't have only one topic for exponential retry; there is a recent change to use only one topic for retries once the max delay has been reached, but each individual retry before that one must be in their own topic; otherwise records ready to process will be blocked behind those that have longer delays.

Ya that make sense. Thanks a lot for your time.

**Reply by hariraogotit** [reply]
> See this answer for how to use a custom naming strategy: https://stackoverflow.com/questions/75723808/send-kafka-retryable-message-to-one-topic-via-spring-retry/75733513#75733513

Yup I tried that.
